.github/workflows/deploy-github-pages-release.yml

```
@@ -0,0 +1,292 @@
+        # get the _templates folder and move it out of the POSYDON folder
+        cp -r $PATH_TO_POSYDON/docs/_source/_templates $GITHUB_WORKSPACE
+
+        # get the injection code from the release
+        injection_code=$(awk '/# INJECTION GRAB START/,/# INJECTION GRAB END/' "${PATH_TO_POSYDON}/docs/_source/conf.py")
+
+        # get the available tags
+        TAGS=$(git tag -l | sort -V | grep "v2" | grep -v "-dev")
+
+        echo "v2 filtered tags:"
+        echo $TAGS
+
+        # build the documentation for each tag
+        for tag in $TAGS
+        do
+          echo "Checking out tag: $tag"
+          git checkout -f $tag
+          git clean -fdx
+          python -m pip install .[doc,vis]
+
+          
+            # copy the _templates folder and inject code only for specific tags
+            if [[ "$tag" == "v2.0.0-pre1" || "$tag" == "v2.0.0-pre2" ]]; then
+              # copy the _templates folder to the docs folder
+              cp -r $GITHUB_WORKSPACE/_templates $PATH_TO_POSYDON/docs
+              # remove the injection code from the conf.py file
+              echo "${injection_code}" >> "${PATH_TO_POSYDON}/docs/conf.py"
+            fi
+
+          # build the documentation
+          cd docs
+          make html
+          touch _build/html/.nojekyll
+          
+          # move and copy docs out of the POSYDON folder
+          tag_without_v=${tag//v}
+          mv _build/html/ $output_folder/$tag_without_v
+
+          # cleanup for the next build
+          make clean
+          cd ../
+        done
+        
+    - name: Deploy to GitHub Pages
+      if: success()
+      uses: crazy-max/ghaction-github-pages@v4
+      with:
+        target_branch: gh-pages
+        build_dir: _build/
+        history: true
+        dry_run: true
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_WORKSPACE: ${{ github.workspace }}
+
+  link_latest:
+    needs: [build_v1, build_v2]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download gh-pages
+      uses: actions/checkout@v4
+      with:
+        ref: gh-pages
+        path: '_build'
+        
+    - name: Find latest version and create symlink
+      run: |
+        cd _build
+        
+        # List all version directories
+        echo "All directories:"
+        ls -la
+        
+        # Find all version directories that match our version pattern
+        VERSIONS=$(ls -d v[0-9]* 2>/dev/null || echo "")
+        
+        if [ -z "$VERSIONS" ]; then
+          echo "No version directories found!"
+          exit 1
+        fi
+        
+        echo "Found versions: $VERSIONS"
+        
+        # First try to find the latest non-pre-release version
+        LATEST=$(echo "$VERSIONS" | grep -v "-pre" | sort -V | tail -n 1)
+        
+        # If no non-pre-release version is found, use the latest including pre-releases
+        if [ -z "$LATEST" ]; then
+          LATEST=$(echo "$VERSIONS" | sort -V | tail -n 1)
+        fi
```

This won't work as:
- grep doesn't like the dash (either leave out the dash or use the `-e` option)
- you will get v1.0.5 now as latest as you only allow the pre if no other version exists.

```suggestion
        # First try to find the overall latest
        LATEST=$(echo "$VERSIONS" | sort -V | tail -n 1)
        
        # Second try to find the latest non-pre-release version
        LATESTNOPRE=$(echo "$VERSIONS" | grep -v "pre" | sort -V | tail -n 1)
        
        # Third get the main version part of $LATEST
        LATESTWITHOUTPRE=$(echo "$LATEST" | awk -F "-pre" '{print $1}')
        
        # If there is a version release after the last pre-release version, use the non pre-release
        if [[ "$LATESTNOPRE" == "$LATESTWITHOUTPRE" ]]; then
          LATEST=$LATESTNOPRE
        fi
```

---

Thank you for catching these and the commit suggestion!
This was a mess to get working and test. So this is great!